gemrb/core/Map.cpp

```diff
@@ -1948,6 +1948,12 @@ unsigned int Map::GetBlocked(unsigned int x, unsigned int y, bool actorsAreBlock
 		return 0;
 	}
 	unsigned int ret = SrchMap[y*Width+x];
+	static NavmapPoint p;
```

Comment:
Why is it static? It will change each call that comes this far.

Reply:
For the same reason why there's a lot of `static Point`s in `FindPath`: I don't want to call the `Point` constructor and destructor every time just because `GetActor` expects a `Point`

Reply:
ah, already optimizing, ok.

Reply:
`Point` is a POD type with a trivial destructor so there won't be any destructor overhead from it. Constructor overhead probably also wouldn't matter when building with optimizations enabled as it should inline to 2 `mov` instructions very easily.

In fact, I would expect the static to _introduce_ overhead since statics are now thread safe constructible in c++11 which will require the addition of an atomic flag check every time control passes over it.

Reply:
Damn, totally forgot about the thread safety of `static` in c++11. 
Also about constructor overhead, of course I was noticing it in the flame graph when profiling the code because profiling == `-O0`. Thanks for the heads up, I'll de-staticize it!

Premature optimization really is the root of all evil.

Reply:
indeed it can be :) I recommend compiler explorer and quick bench if you are ever in doubt. You can see how many more instructions it is by adding and removing static from [my example](https://godbolt.org/z/GuQF8u)

I did lie about `Point` being POD. we wrote a actor for it so its technically not, but that won't stop the compiler from optimizing it just the same. there is also not reason we can't make it POD when we tackle #232. the destructor is trivial still and there will be no overhead from that at all (it won't even exist in the binary).

Reply:
so I was surprised in my linked example to learn that even at `-O3` the compiler wouldn't optimize out the constructor. 2 mov instructions should be smaller and faster than a call and  im pretty sure this qualifies under the as-if rule. Making it POD it was able to [optimize it even better](https://godbolt.org/z/7A566K) than 2 `mov`s. Definitely will be making this change in #232. I checked both gcc and clang latest, neither optimized the store of the non POD and both optimized it down to a single `mov` in the PoD case.

You definitely don't want `state` tho :)

Reply:
Perfect, thank you!